azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/artifacthandler/FTPArtifactHandlerImpl.java

```diff
@@ -32,34 +33,47 @@ public FTPArtifactHandlerImpl(final T mojo) {
     }
 
     protected String getDeploymentStageDirectory() {
-        final String outputFolder = this.mojo.getPluginName().replaceAll(MAVEN_PLUGIN_POSTFIX, "");
-        return Paths.get(mojo.getBuildDirectoryAbsolutePath(), outputFolder, this.mojo.getAppName()).toString();
+        return Paths.get(mojo.getBuildDirectoryAbsolutePath(), this.mojo.getAppName()).toString();
```

Comment:
Why we remove the plugin specific folder?

Reply:
as we synced offline, there is no need to use target/azure-webapp/appname as stage directory, just target/appname is ok "azure-webapp" just hard coded in previous code snippet in ftpartifacthandlerimpl

Reply:
`azure-webapp` is needed here:
- It has no harm to have this folder
- Imaging if in the future the plugin needs to generate some files besides the deployment file, where to place them? The folder `azure-webapp` is to specify the scope for our plugin.
- Take the convention of other popular plugins, for example: `findbugs`

Reply:
convention makes good point here but no harm does not means necessary

files besides deployment files should be treated files that not used during deployment, back to our offline discussion, stage directory is within target folder, it is not visible to users, it is just a temp folder created by plugin to stage the files that will deploy to web app

Reply:
Add another reason that I remove this is that, we move the logic to library module. It is totally ok when we hard code 'azure-webapp' or 'function-app' in their own module, but it is kind of not that natural to split out the plugin name to form a temporary folder invisible to users in target folder in library module. We want keep the dependency is plugin module to library module not the reverse.